varcode/variant_annotator.py

```diff
@@ -40,14 +40,38 @@ def describe_variant(self, variant):
 
         # dictionary from gene ID to list of transcript effects
         gene_transcript_effects_groups = {}
+
+        # mapping from Transcript objects to errors encountered
+        # while trying to annotated them
+        errors = {}
+
         for (gene_id, transcripts) in overlapping_transcript_groups.items():
             effects = []
             for transcript in transcripts:
-                effect = infer_transcript_effect(variant, transcript)
-                effects.append(effect)
+                try:
+                    effects.append(infer_transcript_effect(variant, transcript))
+                except (AssertionError, ValueError) as error:
+                    if raise_on_error:
+                        raise
+                    else:
+                        logging.warn(
+                            "Encountered error annotating %s for %s: %s",
+                            variant,
+                            transcript,
+                            error)
+                    errors[transcript] = error
```

Comment:
Any ideas on what these transcript effects would look like? Would I just have a generic `ErrorEffect` object? Maybe once the class of errors is narrowed down to some well understood types of mistakes in the annotation database I can make those into different `Effect` types, for now I'm just catching `ValueError`s and `AssertionError`s from many different sources and don't have any extra information to unify them other than "something went wrong while trying to annotate this variant".